addons/mail/static/src/js/systray/systray_activity_menu.js

```diff
@@ -131,7 +131,7 @@ var ActivityMenu = Widget.extend({
             res_model:  data.res_model,
             views: [[false, 'kanban'], [false, 'form']],
             search_view_id: [false],
-            domain: [['activity_user_id', '=', session.uid]],
+            domain: [['activity_user_id', '=', session.uid], '|', ['company_id', '=', false], ['company_id', 'child_of', [session.company_id]]],
```

Comment:
Would make layout so that domain is easier to read, and also provide explanation about need of the company part:
```js
domain: [
    ['activity_user_id', '=', session.uid],
    // filter activities on current company
    '|',
    ['company_id', '=', false],
    ['company_id', 'child_of', [session.company_id]]
],
```

It's not super great having to manage multi-company in client-code that way... but that's how it is with current client-side and server-side code, so can't do much better to fix it...